### PR TITLE
[EditableText] honor the "brieflyShowPassword" system setting

### LINKED ISSUE #65352: Flutter app still shows leading password character on Android even after disabling feature in settings (Security Vulnerability)
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows:
```dart
// lib/main.dart
import 'package:flutter/material.dart';

void main() => runApp(App());

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: TextField(obscureText: true),
        ),
      ),
    );
  }
}

```
3. On Android, go to settings.
4. Go to 'Security' or 'Security & Location'
5. Uncheck 'Show Passwords' (Display characters briefly as you type)
6. Open Flutter app `bug` and type in any text

**Expected results:** 
There should be **no** characters displayed for any time period as password 

**Actual results:** 
Characters are **displayed** briefly as you type

<details>
  <summary>Logs</summary>

```
[√] Flutter (Channel stable, 1.20.3, on Microsoft Windows [Version 10.0.10240], locale en-US)
    • Flutter version 1.20.3 at C:\flutter
    • Framework revision 216dee60c0 (6 days ago), 2020-09-01 12:24:47 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\Joseph\AppData\Local\Android\Sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_SDK_ROOT = C:\Users\Joseph\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code, 64-bit edition (version 1.48.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.14.0

[√] Connected device (1 available)
    • Infinix X610B (mobile) • 0425325972000854 • android-arm64 • Android 9 (API 28)

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
- Fixes #65352

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_test/lib/src/window.dart [hunk 1]
```diff
@@ -271,6 +271,15 @@ class TestWindow implements ui.SingletonFlutterWindow {
     platformDispatcher.onTextScaleFactorChanged = callback;
   }
 
+  @override
+  bool get brieflyShowPassword => _brieflyShowPasswordTestValue ?? platformDispatcher.brieflyShowPassword;
+  bool? _brieflyShowPasswordTestValue;
+  /// Hides the real [brieflyShowPassword] and reports the given
+  /// `brieflyShowPassword` instead.
```
Comment (justinmc) [inline packages/flutter_test/lib/src/window.dart:278]: Nit: Maybe say `brieflyShowPasswordTestValue` instead of `brieflyShowPassword`?

**Review by justinmc — COMMENTED** [review]
Quick question in the tests where a value was not what I expected.  Probably I just didn't follow the test correctly.  Otherwise ready to LGTM.

I'm glad we will have this feature now!

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 2]
```diff
@@ -3611,6 +3611,72 @@ void main() {
...
+    await tester.enterText(find.byType(EditableText), 'AA');
+    await tester.pump();
+    await tester.enterText(find.byType(EditableText), 'AAA');
+    await tester.pump();
+
+    tester.binding.window.brieflyShowPasswordTestValue = false;
+    addTearDown(() {
+      tester.binding.window.brieflyShowPasswordTestValue = true;
+    });
+    expect((findRenderEditable(tester).text! as TextSpan).text, '••A');
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/editable_text_test.dart:3671]: it becomes hidden on the next cursor tick.

File: packages/flutter_test/lib/src/window.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter_test/lib/src/window.dart:278]: Ah right good catch!

**Review by justinmc — APPROVED** [review]
LGTM 👍

**Comment by Hixie** [comment]
This was reverted in https://github.com/flutter/flutter/pull/98089

**Pull request merged**